future_canceler.go

```diff
@@ -0,0 +1,17 @@
+package tarantool
+
+type futureCanceler interface {
+	Cancel(fut *Future, err error) error
+}
```

Comment:
The main purpose of the interface is dependency injection that would help write tests. You need to write test for the context & cancel behavior of the Future in `future_test.go`.

Reply:
Added.

Reply:
Added.